File: Problem_Solving/Leet_code_problem_solutions_with_GoLang/1523_odd_count.go
package main

import (
	"fmt"
)

func countOdds(low int, high int) int {
	var diff int
	var count int

	diff = high - low

	if high%2 == 1 && low%2 == 1 {
		count = (diff / 2) + 1
	} else if high%2 == 0 && low%2 == 0 {
		count = int(diff / 2)
	} else {
		count = int(diff/2) + 1
	}
	return count
}

func main() {
	low := 1
	high := 10
	fmt.Println(countOdds(low, high))

}

File: Problem_Solving/Letcode_Problem_Solutions/1523_odd_count.py
import math


class Solution:
    def countOdds(self, low: int, high: int) -> int:
        diff = high - low
        count = 0
        if high % 2 and low % 2:
            count = diff // 2 + 1
        elif high % 2 == 0 and low % 2 == 0:
            count = diff // 2
        else:
            count = diff // 2 + 1
        return count


if __name__ == "__main__":
    s = Solution()
    print(s.countOdds(2, 5))
